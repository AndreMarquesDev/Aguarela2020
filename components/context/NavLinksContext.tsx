import { createContext, Context } from 'react';

export interface NavLinksContextProps {
    setNavHeight?: React.Dispatch<React.SetStateAction<number>>;
}

const NavLinksContext: Context<NavLinksContextProps> = createContext(null);

export default NavLinksContext;
