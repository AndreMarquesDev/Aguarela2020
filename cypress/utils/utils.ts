import { Locale } from '../../types/Locale';
import {
    menuIconButtonDataTestId,
    nukaCarouselNextButtonDataTestId,
} from '../../utils/dataTestIds';
import { TextsInterface, textsPt, textsEn } from '../../utils/texts';

export const getLocalizedTexts = (locale: Locale): TextsInterface => {
    if (locale === Locale.Pt) {
        return textsPt;
    }

    return textsEn;
};

export const openMenuMobile = (isMobile: boolean, isFirefox: boolean): void => {
    if (isMobile) {
        cy.getByDataTestId(menuIconButtonDataTestId).click({ force: isFirefox });
    }
};

export const clickNextArrowButtonIfMobile = (
    isMobile: boolean,
    container: string,
    containerTitle: string
): void => {
    const isFirefox = Cypress.isBrowser('firefox');

    if (isMobile) {
        cy.getByDataTestIdParent(container, nukaCarouselNextButtonDataTestId).click({
            force: isFirefox,
        });

        cy.getByText(container, containerTitle).scrollIntoView();
    }
};
