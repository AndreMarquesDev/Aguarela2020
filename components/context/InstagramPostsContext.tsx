import { createContext, Context } from 'react';
import { IInstagramPost } from '../../utils/generic';

export interface IInstagramPostsContext {
    posts: IInstagramPost[];
}

const InstagramPostsContext: Context<IInstagramPostsContext> = createContext(null);

export default InstagramPostsContext;
