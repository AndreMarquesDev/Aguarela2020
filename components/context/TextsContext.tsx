import { createContext, Context } from 'react';
import { TextsInterface, textsPt } from '../../utils/texts';

export interface TextsContextProps {
    texts: TextsInterface;
}

export const TextsContext: Context<TextsContextProps> = createContext({
    texts: textsPt,
});
