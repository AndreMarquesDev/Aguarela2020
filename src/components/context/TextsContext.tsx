import type { Context } from 'react';
import type { TextsInterface } from '../../utils/texts';
import { createContext } from 'react';
import { textsPt } from '../../utils/texts';

export type TextsContextProps = {
    texts: TextsInterface;
};

export const TextsContext: Context<TextsContextProps> = createContext({
    texts: textsPt,
});
