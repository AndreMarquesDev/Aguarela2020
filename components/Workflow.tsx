import React, { FC, useContext } from 'react';
import TextsContext from './context/TextsContext';
import Title from './Title/Title';

const Workflow: FC = () => {
    const { texts } = useContext(TextsContext);

    return (
        <>
            <section className="container">
                <div className="wrapper genericMargins">
                    <Title text={texts.workflow} />

                    <ul>
                        <li>
                            <p>1.</p>
                            <p>{texts.defineTarget}</p>
                            <div className="backface">
                                <p>{texts.whatIsTheTarget}</p>
                            </div>
                        </li>
                        <li>
                            <p>2.</p>
                            <p>{texts.defineGoals}</p>
                            <div className="backface">
                                <p>{texts.whatAreYourGoals}</p>
                            </div>
                        </li>
                        <li>
                            <p>3.</p>
                            <p>{texts.platformStrategy}</p>
                            <div className="backface">
                                <p>{texts.whatPlatforms}</p>
                            </div>
                        </li>
                        <li>
                            <p>4.</p>
                            <p>{texts.contentCreation}</p>
                            <div className="backface">
                                <p>{texts.relevantContentCreation}</p>
                            </div>
                        </li>
                        <li>
                            <p>5.</p>
                            <p>{texts.paidSocialAndInfluencers}</p>
                            <div className="backface">
                                <p>{texts.promotePostsAndCollaborate}</p>
                            </div>
                        </li>
                        <li>
                            <p>6.</p>
                            <p>{texts.insightsAndReports}</p>
                            <div className="backface">
                                <p>{texts.detailedAnalysis}</p>
                            </div>
                        </li>
                        <li>
                            <p>7.</p>
                            <p>{texts.optimization}</p>
                            <div className="backface">
                                <p>{texts.improvementsAndUpdatesToTheStrategy}</p>
                            </div>
                        </li>
                    </ul>
                </div>
            </section>

            <style jsx>
                {`
                    @import './styles/_vars.scss';

                    $blockSize: 160rem;
                    $blockSizeMobile: 120rem;
                    $backfaceOffset: 12rem;

                    .container {
                        background: $purple;

                        ul {
                            display: flex;
                            justify-content: space-between;

                            @include wrapper {
                                width: 800rem;
                                flex-wrap: wrap;
                                justify-content: center;
                                margin: 0 auto;
                            }

                            @include phablet {
                                width: 600rem;
                            }

                            @include tablet {
                                width: 400rem;
                            }

                            @include mobile {
                                width: 280rem;
                                justify-content: space-between;
                            }

                            li {
                                width: $blockSize;
                                height: $blockSize;
                                display: flex;
                                flex-direction: column;
                                justify-content: center;
                                position: relative;
                                text-align: center;
                                padding: 10rem;
                                z-index: 0;

                                @include wrapper {
                                    margin: 0 calc(7rem + #{$backfaceOffset}) 0 7rem;

                                    &:first-child {
                                        margin-bottom: 50rem;
                                    }
                                }

                                @include phablet {
                                    &:nth-child(4) {
                                        margin-bottom: 50rem;
                                    }
                                }

                                @include tablet {
                                    &:nth-child(5) {
                                        margin-bottom: 50rem;
                                    }
                                }

                                @include mobile {
                                    width: $blockSizeMobile;
                                    height: $blockSizeMobile;
                                    margin: 0;
                                }

                                &:before,
                                &:after {
                                    content: '';
                                    width: $blockSize;
                                    height: $blockSize;
                                    position: absolute;

                                    @include mobile {
                                        width: $blockSizeMobile;
                                        height: $blockSizeMobile;
                                    }
                                }

                                &:before {
                                    top: 0;
                                    left: 0;
                                    background-color: $blue;
                                }

                                &:after {
                                    top: $backfaceOffset;
                                    left: $backfaceOffset;
                                    background-color: $pink;
                                    z-index: -1;
                                }

                                &:hover {
                                    .backface {
                                        display: flex;
                                    }

                                    &:after {
                                        background-color: $blue;
                                    }
                                }

                                p {
                                    @include fontXS($yellow, uppercase, bold);
                                    position: relative;
                                    display: block;
                                    z-index: 0;

                                    @include mobile {
                                        @include fontXXS($yellow, uppercase, bold);
                                    }
                                }

                                .backface {
                                    display: none;
                                    width: 100%;
                                    height: 100%;
                                    position: absolute;
                                    top: 0;
                                    left: 50%;
                                    transform: translateX(-50%);
                                    align-items: center;
                                    background: $pink;
                                    padding: 10rem;

                                    p {
                                        @include fontXS($yellow);

                                        @include mobile {
                                            @include fontXXS($yellow);
                                        }
                                    }
                                }
                            }
                        }
                    }
                `}
            </style>
        </>
    );
};

export default Workflow;
