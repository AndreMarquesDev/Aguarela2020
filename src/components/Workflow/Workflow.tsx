import React, { useContext } from 'react';
import { workflowSectionDataTestId } from '../../utils/dataTestIds';
import { TextsContext } from '../context/TextsContext';
import { Title } from '../Title/Title';
import {
    mobileMediaQuery,
    phabletMediaQuery,
    tabletMediaQuery,
    wrapperMediaQuery,
} from '../../styles/mediaQueries';

export const Workflow = (): JSX.Element => {
    const { texts } = useContext(TextsContext);

    return (
        <>
            <section className="container" data-testid={workflowSectionDataTestId}>
                <div className="wrapper genericMargins">
                    <Title text={texts.workflow} />

                    <ul>
                        <li className="fontXS">
                            <p>1.</p>
                            <p>{texts.defineTarget}</p>
                            <div className="backface">
                                <p>{texts.whatIsTheTarget}</p>
                            </div>
                        </li>
                        <li className="fontXS">
                            <p>2.</p>
                            <p>{texts.defineGoals}</p>
                            <div className="backface">
                                <p>{texts.whatAreYourGoals}</p>
                            </div>
                        </li>
                        <li className="fontXS">
                            <p>3.</p>
                            <p>{texts.platformStrategy}</p>
                            <div className="backface">
                                <p>{texts.whatPlatforms}</p>
                            </div>
                        </li>
                        <li className="fontXS">
                            <p>4.</p>
                            <p>{texts.contentCreation}</p>
                            <div className="backface">
                                <p>{texts.relevantContentCreation}</p>
                            </div>
                        </li>
                        <li className="fontXS">
                            <p>5.</p>
                            <p>{texts.paidSocialAndInfluencers}</p>
                            <div className="backface">
                                <p>{texts.promotePostsAndCollaborate}</p>
                            </div>
                        </li>
                        <li className="fontXS">
                            <p>6.</p>
                            <p>{texts.insightsAndReports}</p>
                            <div className="backface">
                                <p>{texts.detailedAnalysis}</p>
                            </div>
                        </li>
                        <li className="fontXS">
                            <p>7.</p>
                            <p>{texts.optimization}</p>
                            <div className="backface">
                                <p>{texts.improvementsAndUpdatesToTheStrategy}</p>
                            </div>
                        </li>
                    </ul>
                </div>
            </section>

            <style jsx>
                {`
                    .container {
                        --blockSize: 160rem;
                        --blockSizeMobile: 120rem;
                        --backfaceOffset: 12rem;

                        background: var(--purple);

                        ul {
                            display: flex;
                            justify-content: space-between;

                            @media (${wrapperMediaQuery}) {
                                width: 800rem;
                                flex-wrap: wrap;
                                justify-content: center;
                                margin: 0 auto;
                            }

                            @media (${phabletMediaQuery}) {
                                width: 600rem;
                            }

                            @media (${tabletMediaQuery}) {
                                width: 400rem;
                            }

                            @media (${mobileMediaQuery}) {
                                width: 280rem;
                                justify-content: space-between;
                            }

                            li {
                                width: var(--blockSize);
                                height: var(--blockSize);
                                display: flex;
                                flex-direction: column;
                                justify-content: center;
                                position: relative;
                                text-align: center;
                                padding: 10rem;
                                z-index: 0;

                                @media (${wrapperMediaQuery}) {
                                    margin: 0 calc(7rem + var(--backfaceOffset)) 0 7rem;

                                    &:first-child {
                                        margin-bottom: 50rem;
                                    }
                                }

                                @media (${phabletMediaQuery}) {
                                    &:nth-child(4) {
                                        margin-bottom: 50rem;
                                    }
                                }

                                @media (${tabletMediaQuery}) {
                                    &:nth-child(5) {
                                        margin-bottom: 50rem;
                                    }
                                }

                                @media (${mobileMediaQuery}) {
                                    width: var(--blockSizeMobile);
                                    height: var(--blockSizeMobile);
                                    margin: 0;
                                }

                                &:before,
                                &:after {
                                    content: '';
                                    width: var(--blockSize);
                                    height: var(--blockSize);
                                    position: absolute;

                                    @media (${mobileMediaQuery}) {
                                        width: var(--blockSizeMobile);
                                        height: var(--blockSizeMobile);
                                    }
                                }

                                &:before {
                                    top: 0;
                                    left: 0;
                                    background-color: var(--blue);
                                }

                                &:after {
                                    top: var(--backfaceOffset);
                                    left: var(--backfaceOffset);
                                    background-color: var(--pink);
                                    z-index: -1;
                                }

                                &:hover {
                                    .backface {
                                        display: flex;
                                    }

                                    &:after {
                                        background-color: var(--blue);
                                    }
                                }

                                p {
                                    color: var(--yellow);
                                    font-weight: bold;
                                    text-transform: uppercase;
                                    position: relative;
                                    display: block;
                                    z-index: 0;

                                    @media (${mobileMediaQuery}) {
                                        font-size: var(--fontSizeXXS);
                                    }
                                }

                                .backface {
                                    display: none;
                                    width: 100%;
                                    height: 100%;
                                    position: absolute;
                                    top: 0;
                                    left: 50%;
                                    transform: translateX(-50%);
                                    align-items: center;
                                    background: var(--pink);
                                    padding: 10rem;

                                    p {
                                        color: var(--yellow);
                                        font-weight: normal;
                                        text-transform: none;

                                        @media (${mobileMediaQuery}) {
                                            font-size: var(--fontSizeXXS);
                                        }
                                    }
                                }
                            }
                        }
                    }
                `}
            </style>
        </>
    );
};
