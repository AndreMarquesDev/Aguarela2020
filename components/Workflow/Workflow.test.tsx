import '@testing-library/jest-dom';
import React from 'react';
import { render, RenderResult, screen } from '@testing-library/react';
import { Workflow } from './Workflow';
import { TextsContext } from '../context/TextsContext';
import { textsEn, textsPt } from '../../utils/texts';

const renderComponent = (): RenderResult => {
    return render(<Workflow />);
};

describe('<Workflow />', () => {
    test('renders properly', () => {
        const { container } = renderComponent();

        expect(screen.getByText(textsPt.workflow)).toBeInTheDocument();
        expect(screen.getByText(textsPt.defineTarget)).toBeInTheDocument();
        expect(screen.getByText(textsPt.whatIsTheTarget)).toBeInTheDocument();
        expect(screen.getByText(textsPt.defineGoals)).toBeInTheDocument();
        expect(screen.getByText(textsPt.whatAreYourGoals)).toBeInTheDocument();
        expect(screen.getByText(textsPt.platformStrategy)).toBeInTheDocument();
        expect(screen.getByText(textsPt.whatPlatforms)).toBeInTheDocument();
        expect(screen.getByText(textsPt.contentCreation)).toBeInTheDocument();
        expect(screen.getByText(textsPt.relevantContentCreation)).toBeInTheDocument();
        expect(screen.getByText(textsPt.paidSocialAndInfluencers)).toBeInTheDocument();
        expect(screen.getByText(textsPt.promotePostsAndCollaborate)).toBeInTheDocument();
        expect(screen.getByText(textsPt.insightsAndReports)).toBeInTheDocument();
        expect(screen.getByText(textsPt.detailedAnalysis)).toBeInTheDocument();
        expect(screen.getByText(textsPt.optimization)).toBeInTheDocument();
        expect(screen.getByText(textsPt.improvementsAndUpdatesToTheStrategy)).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    test('renders properly in English', () => {
        const { container } = render(
            <TextsContext.Provider
                value={{
                    texts: textsEn,
                }}
            >
                <Workflow />
            </TextsContext.Provider>
        );

        expect(screen.getByText(textsEn.workflow)).toBeInTheDocument();
        expect(screen.getByText(textsEn.defineTarget)).toBeInTheDocument();
        expect(screen.getByText(textsEn.whatIsTheTarget)).toBeInTheDocument();
        expect(screen.getByText(textsEn.defineGoals)).toBeInTheDocument();
        expect(screen.getByText(textsEn.whatAreYourGoals)).toBeInTheDocument();
        expect(screen.getByText(textsEn.platformStrategy)).toBeInTheDocument();
        expect(screen.getByText(textsEn.whatPlatforms)).toBeInTheDocument();
        expect(screen.getByText(textsEn.contentCreation)).toBeInTheDocument();
        expect(screen.getByText(textsEn.relevantContentCreation)).toBeInTheDocument();
        expect(screen.getByText(textsEn.paidSocialAndInfluencers)).toBeInTheDocument();
        expect(screen.getByText(textsEn.promotePostsAndCollaborate)).toBeInTheDocument();
        expect(screen.getByText(textsEn.insightsAndReports)).toBeInTheDocument();
        expect(screen.getByText(textsEn.detailedAnalysis)).toBeInTheDocument();
        expect(screen.getByText(textsEn.optimization)).toBeInTheDocument();
        expect(screen.getByText(textsEn.improvementsAndUpdatesToTheStrategy)).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });
});
