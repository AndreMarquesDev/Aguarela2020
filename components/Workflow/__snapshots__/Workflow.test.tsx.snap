// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Workflow /> renders properly 1`] = `
<div>
  <section
    class="jsx-3904647779 container"
  >
    <div
      class="jsx-3904647779 wrapper genericMargins"
    >
      <h1
        class="jsx-3155756322 "
      >
        Workflow
      </h1>
      <ul
        class="jsx-3904647779"
      >
        <li
          class="jsx-3904647779"
        >
          <p
            class="jsx-3904647779"
          >
            1.
          </p>
          <p
            class="jsx-3904647779"
          >
            Definir target
          </p>
          <div
            class="jsx-3904647779 backface"
          >
            <p
              class="jsx-3904647779"
            >
              Quem é o público-alvo da marca
            </p>
          </div>
        </li>
        <li
          class="jsx-3904647779"
        >
          <p
            class="jsx-3904647779"
          >
            2.
          </p>
          <p
            class="jsx-3904647779"
          >
            Definir objetivos
          </p>
          <div
            class="jsx-3904647779 backface"
          >
            <p
              class="jsx-3904647779"
            >
              Quais são os objetivos de negócio e comunicação
            </p>
          </div>
        </li>
        <li
          class="jsx-3904647779"
        >
          <p
            class="jsx-3904647779"
          >
            3.
          </p>
          <p
            class="jsx-3904647779"
          >
            Estratégia de plataformas
          </p>
          <div
            class="jsx-3904647779 backface"
          >
            <p
              class="jsx-3904647779"
            >
              Quais as plataformas onde a marca deve estar presente
            </p>
          </div>
        </li>
        <li
          class="jsx-3904647779"
        >
          <p
            class="jsx-3904647779"
          >
            4.
          </p>
          <p
            class="jsx-3904647779"
          >
            Criação de Conteúdo
          </p>
          <div
            class="jsx-3904647779 backface"
          >
            <p
              class="jsx-3904647779"
            >
              Criação de conteúdo relevante e de qualidade para o target
            </p>
          </div>
        </li>
        <li
          class="jsx-3904647779"
        >
          <p
            class="jsx-3904647779"
          >
            5.
          </p>
          <p
            class="jsx-3904647779"
          >
            Paid social & influencers
          </p>
          <div
            class="jsx-3904647779 backface"
          >
            <p
              class="jsx-3904647779"
            >
              Promoção de posts e colaborações com influencers
            </p>
          </div>
        </li>
        <li
          class="jsx-3904647779"
        >
          <p
            class="jsx-3904647779"
          >
            6.
          </p>
          <p
            class="jsx-3904647779"
          >
            Insights e reports
          </p>
          <div
            class="jsx-3904647779 backface"
          >
            <p
              class="jsx-3904647779"
            >
              Análises detalhadas e identificação de insights
            </p>
          </div>
        </li>
        <li
          class="jsx-3904647779"
        >
          <p
            class="jsx-3904647779"
          >
            7.
          </p>
          <p
            class="jsx-3904647779"
          >
            Otimização
          </p>
          <div
            class="jsx-3904647779 backface"
          >
            <p
              class="jsx-3904647779"
            >
              Melhorias e updates à estratégia
            </p>
          </div>
        </li>
      </ul>
    </div>
  </section>
</div>
`;

exports[`<Workflow /> renders properly in English 1`] = `
<div>
  <section
    class="jsx-3904647779 container"
  >
    <div
      class="jsx-3904647779 wrapper genericMargins"
    >
      <h1
        class="jsx-3155756322 "
      >
        Workflow
      </h1>
      <ul
        class="jsx-3904647779"
      >
        <li
          class="jsx-3904647779"
        >
          <p
            class="jsx-3904647779"
          >
            1.
          </p>
          <p
            class="jsx-3904647779"
          >
            Set the target
          </p>
          <div
            class="jsx-3904647779 backface"
          >
            <p
              class="jsx-3904647779"
            >
              What's the target
            </p>
          </div>
        </li>
        <li
          class="jsx-3904647779"
        >
          <p
            class="jsx-3904647779"
          >
            2.
          </p>
          <p
            class="jsx-3904647779"
          >
            Set goals
          </p>
          <div
            class="jsx-3904647779 backface"
          >
            <p
              class="jsx-3904647779"
            >
              What are the business and communication goals
            </p>
          </div>
        </li>
        <li
          class="jsx-3904647779"
        >
          <p
            class="jsx-3904647779"
          >
            3.
          </p>
          <p
            class="jsx-3904647779"
          >
            Platforms strategies
          </p>
          <div
            class="jsx-3904647779 backface"
          >
            <p
              class="jsx-3904647779"
            >
              In what platforms should the brand be present
            </p>
          </div>
        </li>
        <li
          class="jsx-3904647779"
        >
          <p
            class="jsx-3904647779"
          >
            4.
          </p>
          <p
            class="jsx-3904647779"
          >
            Content creation
          </p>
          <div
            class="jsx-3904647779 backface"
          >
            <p
              class="jsx-3904647779"
            >
              Relevant and quality content created for the target
            </p>
          </div>
        </li>
        <li
          class="jsx-3904647779"
        >
          <p
            class="jsx-3904647779"
          >
            5.
          </p>
          <p
            class="jsx-3904647779"
          >
            Paid social & influencers
          </p>
          <div
            class="jsx-3904647779 backface"
          >
            <p
              class="jsx-3904647779"
            >
              Promoted posts and collaborations with influencers
            </p>
          </div>
        </li>
        <li
          class="jsx-3904647779"
        >
          <p
            class="jsx-3904647779"
          >
            6.
          </p>
          <p
            class="jsx-3904647779"
          >
            Insights and reports
          </p>
          <div
            class="jsx-3904647779 backface"
          >
            <p
              class="jsx-3904647779"
            >
              Detailed analysis and insights' finding
            </p>
          </div>
        </li>
        <li
          class="jsx-3904647779"
        >
          <p
            class="jsx-3904647779"
          >
            7.
          </p>
          <p
            class="jsx-3904647779"
          >
            Optimization
          </p>
          <div
            class="jsx-3904647779 backface"
          >
            <p
              class="jsx-3904647779"
            >
              Strategy updates and improvements
            </p>
          </div>
        </li>
      </ul>
    </div>
  </section>
</div>
`;
